### ISSUE TITLE
Using `struct` type as an argument of `paramscan()` function for faster simulation

### ISSUE THREAD

**Issue by yunhwankim2** [JuliaDynamics/Agents.jl#1107]
In the [Performance Tips section](https://juliadynamics.github.io/Agents.jl/stable/performance_tips/#Use-Type-stable-containers-for-the-model-properties) of the documentation, it is recommended to use `struct` for model properties. 
However, when I tried to use `paramscan()` for simulation, the function requires to input a parameter set in the `AbstractDict` type, which makes the simulation slower.
How can I use `struct` type as an argument of `paramscan()` function? I think it would make the simulation faster.
Thank you.

**Comment by Datseris** [comment]
Yeah, you can't use `paramscan` with a `struct` container because it uses a dict to make a combinatorial expansion into new dicts. You would need to make a pull request that expands the `paramscan` function so that it can create custom structs instead of dicts given the type of struct to make. This is probably a rather simple change that would take 5 lines of code or so. You would have to modify `dict_list` to have 2 inputs, the second being the type of output (instead of the default dict).

If you don't want to do this, then you can instead copy/paste the following lines from the source code of `paramscan`:

https://github.com/JuliaDynamics/Agents.jl/blob/24bd774f55e86cfc0b6b2b4f4a7cb35b0abe8592/src/simulations/paramscan.jl#L103-L115

`combs` is a vector of all parameter-structs you want to loop your simulation over.

**Comment by yunhwankim2** [comment]
I'm sorry but it seems not simple to me :). My programming skill is not good enough yet.
But I'll keep studying the code to find anything that I can contribute.
Thank you for the reply.
